buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java

```diff
@@ -95,24 +96,34 @@ private static void setupRootJdkDownload(Project rootProject, String platform, S
 
         // add fake ivy repo for jdk url
         String repoName = "jdk_repo_" + version;
+        RepositoryHandler repositories = rootProject.getRepositories();
         if (rootProject.getRepositories().findByName(repoName) == null) {
-            // simpler legacy pattern from JDK 9 to JDK 12 that we are advocating to Oracle to bring back
-            rootProject.getRepositories().ivy(ivyRepo -> {
-                ivyRepo.setName(repoName);
-                ivyRepo.setUrl("https://download.oracle.com");
-                ivyRepo.metadataSources(IvyArtifactRepository.MetadataSources::artifact);
-                ivyRepo.patternLayout(layout ->
-                    layout.artifact("java/GA/jdk" + jdkMajor + "/" + jdkBuild + "/GPL/openjdk-[revision]_[module]-x64_bin.[ext]"));
-                ivyRepo.content(content -> content.includeGroup("jdk"));
-            });
-            // current pattern since 12.0.1
-            rootProject.getRepositories().ivy(ivyRepo -> {
-                ivyRepo.setName(repoName + "_with_hash");
-                ivyRepo.setUrl("https://download.oracle.com");
-                ivyRepo.metadataSources(IvyArtifactRepository.MetadataSources::artifact);
-                ivyRepo.patternLayout(layout -> layout.artifact(
-                    "java/GA/jdk" + jdkVersion + "/" + hash + "/" + jdkBuild + "/GPL/openjdk-[revision]_[module]-x64_bin.[ext]"));
-                ivyRepo.content(content -> content.includeGroup("jdk"));
+            if (Integer.parseInt(jdkMajor) >= 12 && jdkVersion.equals("12") == false) {
```

Comment:
Just so I'm following, a `jdkVersion` here would be "12" in the case of 12.0.0 and I guess "12.0.1" for the new pattern?

Reply:
Alternatively could we not just do `if (hash != null)`?

Reply:
Yes that is correct. You are right we could just check for the hash. I will change to that.